Permet de regarder les stats depuis la première et la dernière publi

artragis opened the pull request:
Implement #5165 

Pour la QA

- publiez plusieurs fois un tuto 
- regarder les stats sur les menu "depuis la dernière publication" et "depuis la première publication"

peut-être passer par la béta pour tester  ça @Situphen ?

artragis committed:
Permet de regarder les stats depuis la première et la dernière publication

Situphen commented:
Tu veux que je le mette sur la bêta ce soir ?

artragis commented:
ça serais chouette en effet.

Situphen commented:
On est tellement synchronisés, j'étais justement en train de mettre ça en bêta ! :D 

Je viens d'essayer et je peux à minima dire qu'il y a un petit problème pour la mise en gras des boutons : 

![Aperçu du problème](https://user-images.githubusercontent.com/6664636/51550186-2d7d5300-1e6c-11e9-8e85-7b64f3f3cc5c.png)

artragis commented:
oui et non y'a un problème car "start_date" n'est pas bien setté. Je regarderai demain en espérant avoir un peu plus de temps qu'aujourd'hui.

artragis commented:
J'ai fait les modifs. Chez moi ça marche.

Pour la mise en gras c'est... attendu. En fait quand première et dernière publication sont identique, les liens sont les mêmes alors le système les met en gras.

Situphen commented:
Ah d'accord ! Bon, il faut trouver un moyen pour expliquer ça simplement sinon on va avoir des rapports de bugs pour ça :)

artragis commented:
J'ai dit que c'était "attendu", pas que c'était "normal", je vais voir si on ne peut pas mettre un des liens en grisé s'ils sont identiques

Situphen commented:
Je pense qu'il faut éviter la présence des deux liens, il faut un seul lien "Depuis la publication" quand il n'y a eu qu'une seule publication :)

artragis commented:
Pas faux.

artragis commented:
et voilà.

Situphen commented on templates/tutorialv2/stats/index.html:
```diff
@@ -182,6 +182,24 @@ <h3>{% trans "Échelle de temps" %}</h3>
+                    </a>
+                    {% endwith %}
+                </li>
+                {% if db_content.first_publication_date.date != public_version.publication_date.date %}
```
J'ai mis à jour la bêta mais ça ne fonctionne pas, je pense avoir trouvé l'erreur :) (j'ai comparé avec la ligne 186)

```suggestion
                {% if db_content.first_publication_date.date != db_content.public_version.publication_date.date %}
```

Situphen committed:
Corrige la résolution des dates

- on utilise les objets de validation pour la first_publication_date
- on utilise le max(update_date, publication_date) pour la last_publication

Co-Authored-By: artragis <artragis@gmail.com>

artragis commented:
Merci @Situphen j'avais fait un copier foirer :)

Situphen commented:
C'est sur la bêta :)

artragis commented:
Hum il semble toujours y avoir un bug pour la première date de publication.

artragis commented:
C'est bon ça marche et j'ai pu bien tester ça chez moi.

A-312 commented:
Ça ne serait pas mieux de mettre "Depuis une publication" et d'ouvrir une modal qui propose le choix d'une publication ?

Je propose ça mais je ne me rends pas compte des modifications nécessaires.

artragis commented:
> Ça ne serait pas mieux de mettre "Depuis une publication" et d'ouvrir une modal qui propose le choix d'une publication ?

ce n'est pas ce qui a été demandé par les auteurs. la première et la dernière c'est vraiment celles qui ont du sens.

Situphen commented:
J'ai testé avec [ce contenu](https://beta.zestedesavoir.com/contenus/99/zeste-de-savoir-passe-en-version-1-7/) sur la bêta et ce n'est pas bon ! Ce contenu a été publié le 24 avril 2015 et a reçu une mise à jour (publication mineure je pense) le 29 octobre 2015 (version actuellement en ligne). J'ai : 

- Depuis la dernière publication : du 24 avril 2015 à aujourd'hui
- Depuis la première publication : du 29 octobre 2015 à aujourd'hui

C'est inversé.

artragis commented:
J'ai essayé avec celui-ci https://beta.zestedesavoir.com/contenus/stats/597/creez-votre-site-web-avec-asp-net/ et ça marche mieux.

Je **pense** qu'en 2015 on avait des problèmes avec les dates de maj & co et que c'est ça qui a fait foirer dans le cas de l'article : il n'y a qu'une validation qui a le statu "accept" et il date du 29/10/2015

Situphen commented:
Je n'ai pas les droits Staff donc impossible pour moi de vérifier sur des contenus récents. Tu as vérifié que ça marchait sur d'autres contenus et le code m'a l'air correct. Je propose qu'on merge et qu'on corrige plus tard si jamais on se rend compte que le problème apparaît sur certains contenus récents. Tu en penses quoi ?

artragis commented:
j'ai essayé sur plusieurs tutos qui ont connu plusieurs publication, comme le tuto c++ par exemple.

Situphen commented:
Parfait, merci pour cette PR @artragis !
